lib/node_modules/@stdlib/math/base/special/cinv/src/main.c

```diff
@@ -0,0 +1,99 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/cinv.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/constants/float64/max.h"
+#include "stdlib/constants/float64/eps.h"
+#include "stdlib/constants/float64/smallest_normal.h"
+#include "stdlib/complex/float64.h"
+#include "stdlib/complex/reim.h"
+
+
+// VARIABLES //
+
+const double LARGE_THRESHOLD = STDLIB_CONSTANT_FLOAT64_MAX * 0.5;
+const double SMALL_THRESHOLD = STDLIB_CONSTANT_FLOAT64_SMALLEST_NORMAL * ( 2.0 / STDLIB_CONSTANT_FLOAT64_EPS );
+const double RECIP_EPS_SQR = 2.0 / ( STDLIB_CONSTANT_FLOAT64_EPS * STDLIB_CONSTANT_FLOAT64_EPS );
+
+
+// MAIN //
+
+/**
+Computes the inverse of a double-precision complex floating-point number.
+*
+* ## References
+*
+* -   Baudin, Michael, and Robert L. Smith. 2012. "A Robust Complex Division in Scilab." _arXiv_ abs/1210.4539 \[cs.MS\] (October): 1–25. <https://arxiv.org/abs/1210.4539>.
+*
+* @param z       input value
+* @return        result
+*
+* @example
+* #include "stdlib/complex/float64.h"
+* #include "stdlib/complex/real.h"
+* #include "stdlib/complex/imag.h"
+*
+* stdlib_complex128_t z = stdlib_complex128( 2.0, 4.0 );
+*
+* stdlib_complex128_t out = stdlib_base_cinv( z );
+*
+* double re = stdlib_real( out );
+* // returns 0.1
+*
+* double im = stdlib_imag( out );
+* // returns -0.2
+*/
+stdlib_complex128_t stdlib_base_cinv( const stdlib_complex128_t z ) {
+	double re;
+	double im;
+	double ab;
+	double s;
+	double r;
+	double t;
+
+	stdlib_reim( z, &re, &im );
+
+	ab = stdlib_base_abs( re ) > stdlib_base_abs( im ) ?
```

Comment:
Duplicate invocation of `abs` is not particularly desirable. Let's go ahead and revert to using `fmax` with a TODO that we should replace once stdlib equivalent.

Reply:
Doesn't the C compiler catch the duplicate calls? Do you prefer to add the `fmax` or to create two variables to avoid calling two times?

Reply:
Re: C compiler. What compiler optimization are you thinking that the C compiler can perform when seeing duplicate calls?

Regarding changes, I'd say to revert to using `fmax`. Creating additional variables seems, to me at least, a bit more clunky.

Reply:
I found some mixed opinions on different forums. Some people state that calls with the same arguments **may** be optimized by the compiler given that in this case it is within a basic block. But apparently that depends as well if the function is pure or impure. So the answer is maybe (?)

I revert the changes anyway (:

Reply:
> So the answer is maybe

Yeah, that is my impression. :) The main compiler optimization would be constant folding/propagation. This is doable for intrinsic functions, but less likely for user-land functions, as harder to guarantee that a function is pure (e.g., `abs` could return "random" values). My personal take is that we should leave little to chance and not overly rely on compiler magic. The "correct" thing to do here is use a `max` function. As we don't have a C implementation (yet), the thing to do is use `fmax` as you originally did and add a TODO.